src/celengine/astrocat.cpp

```diff
@@ -0,0 +1,67 @@
+
+#include <cstdio>
+
+#include "astrocat.h" 
+
+AstroCatalog::IndexNumber SimpleAstroCatalog::getIndexNumberByName(const std::string &name)
+{
+    if (compareIgnoringCase(name, m_prefix, m_prefix.length()) == 0)
+    {
+        IndexNumber num;
+        // Use scanf to see if we have a valid catalog number; it must be
+        // of the form: <prefix> <non-negative integer>  No additional
+        // characters other than whitespace are allowed after the number.
+        char extra[4];
+        if (std::sscanf(name.c_str() + m_prefix.length(), " %u %c", &num, extra) == 1)
```

Comment:
what if you have 2 spaces?

Reply:
space before %u is not required

Reply:
This code is mostly copied from `stardb.cpp`. I don't know how to handle mutliple spaces. The same for others cases with `sscanf`.

Reply:
I incorrectly read the spec, I have thought that whitespace can be input for %c